### ISSUE TITLE
Inconsistent colors on rectangular images

### ISSUE THREAD

**Issue by joelphilippage** [mkaflowski/Media-Style-Palette#1]
When passing a rectangular image to the Media-Style-Notification and the app, I get inconsistent colors. I am guessing the notification center crops the image to a square before looking for colors. Could this be added as an option?

**Comment by mkaflowski** [comment]
I have just copied and modified code from AOSP. Do you use exactly the same bitmap for notification and for the lib? I will test it.

**Comment by joelphilippage** [comment]
Yeah. Here is my code for loading the information on the notification:
`Glide.with(this)
                                .asBitmap()
                                .error(requestBuilder)
                                .load(book.getCover())
                                .into(new SimpleTarget<Bitmap>() {
                                    @Override
                                    public void onResourceReady(Bitmap largeIcon, Transition transition) {
                                        Cat.d("Finished loading");
                                        cover = largeIcon;
                                        // initBuilder
                                        builder.setLargeIcon(largeIcon);
                                        mediaSession.setMetadata(new MediaMetadataCompat.Builder()
                                                .putBitmap(MediaMetadataCompat.METADATA_KEY_ALBUM_ART, largeIcon)
                                                .putString(MediaMetadataCompat.METADATA_KEY_ARTIST, book.getTitle())
                                                .putString(MediaMetadataCompat.METADATA_KEY_TITLE, book.getAuthor())
                                                .build());
                                        startNotification();
                                    }
                                });`
And here it is for passing the image to MediaStylePalette within the app:
`GlideApp.with(context)
                .asBitmap()
                .load(this.getCover())
                .into(new SimpleTarget<Bitmap>() {
                    @Override
                    public void onResourceReady(@NonNull Bitmap bitmap, @Nullable Transition<? super Bitmap> transition) {
                        MediaNotificationProcessor processor =
                                new MediaNotificationProcessor(context, bitmap);
                        final int primaryColor = processor.getBackgroundColor();
                        final int accentColor = processor.getPrimaryTextColor();
                        final int strippedAccent = processor.getPrimaryTextColor();`

Here is an example image from the app:
http://turndapage.com/wp-content/uploads/2018/06/device-2018-06-05-091750.png
And here is the corresponding notification:
http://turndapage.com/wp-content/uploads/2018/06/device-2018-06-05-091814.png
And here is the app after cropping the bitmap to a square. This might not be very accurate since I am cropping to 400x400 instead of the width of the image, but the colors match much better.
http://turndapage.com/wp-content/uploads/2018/06/playback_screenshot.png

**Comment by mkaflowski** [comment]
It looks like the crop cause. I will try to fix that in couple days.

**Comment by mkaflowski** [comment]
I still get right results:
https://ibb.co/mgZu8T
https://ibb.co/nDOAho
Could you send me your image (I cropped this from your screens).
I think that Glide could return different images (it uses some compression I think).

**Comment by joelphilippage** [comment]
I am using a url with glide: https://s-media-cache-ak0.pinimg.com/originals/56/31/bd/5631bda81f2d499283a288106f1b0dff.jpg. The app lets you pick an image from online and loads it from there.
